view/frontend/web/js/checkout-button-product.js

```diff
@@ -198,6 +198,12 @@ function ($, Component, idsResolver, productInfoResolver, _) {
                         'form': $form,
                         'response': res
                     });
+
+                    self.$checkoutButton.dispatchEvent(new CustomEvent('amwalPreCheckoutTriggerError', {
+                        detail: {
+                            description: res?.responseJSON?.message
+                        }
+                    }));
```

Comment:
If you include `amwalErrorHandler` in this component as well you can use the error handler here too

```suggestion
                    amwalErrorHandler.process(self.checkoutButton, res?.responseJSON?.message, 'amwalPreCheckoutTriggerError');
```

Reply:
not sure if I did everything correctly, but I did it similar to the other file. Is there a documentation on this Javascript framework?